app/Filament/Resources/CollectionResource.php

```diff
@@ -87,9 +88,16 @@ public static function table(Table $table): Table
                 ActionGroup::make([
                     Action::make('updateIsFeatured')
                         ->action(function (Collection $collection) {
-                            $collection->update([
-                                'is_featured' => ! $collection->is_featured,
-                            ]);
+                            if (! $collection->is_featured && Collection::where('is_featured', true)->count() >= 4) {
+                                Notification::make()
```

Comment:
PR is ok but I think we should add a scope for featured collections and replace all references to `where('is_featured'` with `Collection::featured()`